caikit/runtime/train.py

```diff
@@ -44,10 +45,34 @@
 log = alog.use_channel("TRAIN")
 error = error_handler.get(log)
 
+USER_ERROR_EXIT_CODE = 1
+INTERNAL_ERROR_EXIT_CODE = 200
+
+
+class ArgumentParserError(Exception):
+    """Custom exception class for ArgumentParser errors."""
+
+
+class TrainArgumentParser(argparse.ArgumentParser):
+    def error(self, message):
+        """Error handler that raises an exception instead of exiting."""
+        raise ArgumentParserError(f"{self.prog}: error: {message}")
+
+
+def write_termination_log(text: str, log_file="/dev/termination-log"):
+    try:
+        with open(log_file, "a") as handle:
+            handle.write(text)
+    except Exception as e:
+        log.warning(
+            "<COR96300323W>",
+            "Unable to write termination log due to error {}".format(e),
+        )
```

Comment:
```suggestion
            "Unable to write termination log due to error %s",
            e,
        )
```

Reply:
This is actually pretty important as a general practice. In this instance, it doesn't really matter because it's behind an `except`, but in general we need to make sure that all log messages use lazy interpolation (note the `,` instead of `%` between the format string and the args). One of the key tenants of `alog` is that log messages should cost nothing more than an `if` statement when disabled. If we do proactive interpolation, the log message must be created before the `if` statement is evaluated which is extremely wasteful. In the worst example I've seen, a single `%` -> `,` saved almost 25% performance of a specific `watson_nlp` module because it was buried down in a deeply nested loop.